src/collection/tx_data/historical_data_collector.py

```diff
@@ -0,0 +1,78 @@
+"""
+Collector of raw transaction data for historical transactions.
+Dependent on `transactions` db table as its source of blocks to fetch. All PPKs should be known.
+
+
+Fetch all transactions from this block, filter by PPKs of all protocols (i.e., Solend, Mango, etc).
+
+1. Obtains several (`BATCH_SIZE`) oldest slot numbers where there are relevant transactions
+ that do not have transaction data stored in database.
+2. Concurrently fetch blocks from these slots.
+3. Filter all transactions in fetched blocks by PPKs of all lending protocols to select relevant transactions.
+4. For each relevant transaction:
+    - create new record in transactions table if not yet exists.
+    - assign transaction data to existing records.
+5. Repeat 1-4 indefinitely.
```

Comment:
Well, repeat until all transactions are completed, right? :slightly_smiling_face:

Reply:
it does not stop ever. it will check database for missing transactions data again and again. In case if we still collecting signatures, we might recieve more records with missing data. 
Otherwise, we can turn it of manually. Should we implement some break points? it might be tricky and probably timeout based, like "if (30 mins no missing transactions data) -> shut down collection"

Reply:
You're right! Let's keep it running forever .. but we need to make sure to decrease the rate limit once the historical collection is completed so that we can use the limits somewhere else, or use a smaller subscription.